plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/types/ClassEmbeddingDetails.kt

```diff
@@ -6,12 +6,14 @@
 package org.jetbrains.kotlin.formver.embeddings.types
 
 import org.jetbrains.kotlin.formver.embeddings.FieldEmbedding
+import org.jetbrains.kotlin.formver.embeddings.expression.OperatorExpEmbeddings.StringLength
 import org.jetbrains.kotlin.formver.names.*
 import org.jetbrains.kotlin.formver.viper.ast.PermExp
 import org.jetbrains.kotlin.formver.viper.ast.Predicate
+import org.jetbrains.kotlin.name.Name
 import org.jetbrains.kotlin.utils.addToStdlib.ifTrue
 
-class ClassEmbeddingDetails(val type: ClassTypeEmbedding, val isInterface: Boolean) : TypeInvariantHolder {
+open class ClassEmbeddingDetails(val type: ClassTypeEmbedding, val isInterface: Boolean) : TypeInvariantHolder {
```

Comment:
I am not sure that inheritance is a good idea here.  While we are in a situation now where the special cases are all dijsoint, perhaps they could be given some other way, like via constructor parameters or using a method call?

Reply:
We're redefining an implementation of one of the methods. It seems kind of like inheritance...
We could do sealed base and default / string implementation. Anyway, I don't like how I inherit from the extension function of some builder...

Reply:
I think the key point is that you aren't really specialising the whole thing, so using composition (with something that might use inheritance itself) would feel more natural/clean to me.